modules/core/04-channel/keeper/grpc_query_test.go

```diff
@@ -1539,3 +1540,95 @@ func (suite *KeeperTestSuite) TestQueryUpgradeError() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestQueryUpgrade() {
+	var (
+		req             *types.QueryUpgradeRequest
+		expectedUpgrade types.Upgrade
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"empty request",
+			func() {
+				req = nil
+			},
+			false,
+		},
+		{
+			"invalid port ID",
+			func() {
+				req = &types.QueryUpgradeRequest{
+					PortId:    "",
+					ChannelId: "test-channel-id",
+				}
+			},
+			false,
+		},
+		{
+			"invalid channel ID",
+			func() {
+				req = &types.QueryUpgradeRequest{
+					PortId:    "test-port-id",
+					ChannelId: "",
+				}
+			},
+			false,
+		},
+		{
+			"channel not found",
+			func() {
+				req = &types.QueryUpgradeRequest{
+					PortId:    "test-port-id",
+					ChannelId: "test-channel-id",
+				}
+			},
+			false,
+		},
```

Comment:
should we add one more test for querying an upgrade on a channel that hasn't initiated an upgrade yet?

Reply:
this test is actually just directly setting the upgrade in state, the upgrade isn't being set in the ChanUpgradeInit function

Reply:
actually we can still do this by just deleting the key 👍